server/command.go

```diff
@@ -0,0 +1,409 @@
+		return &model.CommandResponse{}
+	}
+	if len(args) == 0 {
+		response := executeConfluenceDefault(p, context, args...)
+		return response
+	}
+	confluenceURL, instance, err := p.installCloudInstance(args[0])
+	if err != nil {
+		return p.responsef(context, err.Error())
+	}
+
+	return p.respondCommandTemplate(context, "/command/install_cloud.md", map[string]string{
+		"ConfluenceURL": confluenceURL,
+		"RedirectURL":   instance.GetRedirectURL(),
+		"CloudURL":      p.GetPluginURL() + p.GetAtlassianConnectURLPath(),
+	})
+}
+
+func showInstallServerHelp(p *Plugin, context *model.CommandArgs, args ...string) *model.CommandResponse {
+	if !utils.IsSystemAdmin(context.UserId) {
+		p.postCommandResponse(context, installOnlySystemAdmin)
+		return &model.CommandResponse{}
+	}
+	if len(args) == 0 {
+		response := executeConfluenceDefault(p, context, args...)
+		return response
+	}
+	confluenceURL, instance, err := p.installServerInstance(args[0])
+	if err != nil {
+		return p.responsef(context, err.Error())
+	}
+
+	return p.respondCommandTemplate(context, "/command/install_server.md", map[string]string{
+		"ConfluenceURL": confluenceURL,
+		"RedirectURL":   instance.GetRedirectURL(),
+	})
+}
+
+func deleteSubscription(p *Plugin, context *model.CommandArgs, args ...string) *model.CommandResponse {
+	if len(args) == 0 {
+		p.postCommandResponse(context, specifyAlias)
+		return &model.CommandResponse{}
+	}
+	alias := strings.Join(args, " ")
+	if err := p.DeleteSubscription(context.ChannelId, alias, context.UserId); err != nil {
+		p.postCommandResponse(context, err.Error())
+		return &model.CommandResponse{}
+	}
+	p.postCommandResponse(context, fmt.Sprintf(subscriptionDeleteSuccess, alias))
+	return &model.CommandResponse{}
+}
+
+func listChannelSubscription(p *Plugin, context *model.CommandArgs, args ...string) *model.CommandResponse {
+	channelSubscriptions, gErr := service.GetSubscriptionsByChannelID(context.ChannelId)
+	if gErr != nil {
+		p.postCommandResponse(context, gErr.Error())
+		return &model.CommandResponse{}
+	}
+
+	if len(channelSubscriptions) == 0 {
+		p.postCommandResponse(context, noChannelSubscription)
+		return &model.CommandResponse{}
+	}
+	list := serializer.FormattedSubscriptionList(channelSubscriptions)
+	p.postCommandResponse(context, list)
+	return &model.CommandResponse{}
+}
+
+func confluenceHelpCommand(p *Plugin, context *model.CommandArgs, args ...string) *model.CommandResponse {
+	helpText := getFullHelpText(p, context, args...)
+
+	p.postCommandResponse(context, helpText)
+	return &model.CommandResponse{}
+}
+
+func getFullHelpText(p *Plugin, context *model.CommandArgs, args ...string) string {
+	helpText := commonHelpText
+	if utils.IsSystemAdmin(context.UserId) {
+		helpText += sysAdminHelpText
+	}
+	return helpText
+}
+
+func executeConnect(p *Plugin, context *model.CommandArgs, args ...string) *model.CommandResponse {
+	instances, err := p.instanceStore.LoadInstances()
+	if err != nil {
```

Comment:
I think this could have some of the logic outside so we could improve testability and make it easier to review, WDYT?

An example would be to wrap the userinfo logic into a method/function and the instance logic (from 329 to 338). These are the first things that come to mind.